deployments/cognito-rds-s3/terraform/main.tf

```diff
@@ -1,21 +1,28 @@
 locals {
-  name = "rds-s3-kubeflow"
-  cluster_name = coalesce(var.cluster_name, local.name)
+  cluster_name = var.cluster_name
   region       = var.cluster_region
   eks_version = var.eks_version
 
   vpc_cidr = "10.0.0.0/16"
-  az_count = local.region == "us-west-1" ? 2 : 3
-  azs      = slice(data.aws_availability_zones.available.names, 0, local.az_count)
+
+  # the ordering of the aws_ec2_instance_type_offerings result changes for each query
+  # since we take the first few elements of the result, this causes new AZs to be 
+  # chosen for each terraform apply, which is not desirable
+  available_azs = tolist(setintersection(data.aws_availability_zones.available.names, data.aws_ec2_instance_type_offerings.availability_zones.locations))
+
+  az_count = min(length(local.available_azs), 3)
+  azs      = slice(local.available_azs, 0, local.az_count)
 
   tags = {
-    Blueprint  = local.name
+    Blueprint  = local.cluster_name
     GithubRepo = "github.com/awslabs/kubeflow-manifests"
     Platform = "kubeflow-on-aws"
     KubeflowVersion = "1.6"
```

Comment:
```suggestion
    Blueprint              = local.cluster_name
    GithubRepo         = "github.com/awslabs/kubeflow-manifests"
    Platform.              = "kubeflow-on-aws"
    KubeflowVersion = "1.6"
```

Reply:
Will add a terraform formatter and submit a formatting pr separately since it will be formatting almost every file